src/Infrastructure/LeanCode.Logging/IHostBuilderExtensions.cs

```diff
@@ -83,10 +92,18 @@ public static IHostBuilder ConfigureDefaultLogging(
 
                 additionalLoggingConfiguration?.Invoke(context, loggerConfiguration);
 
-                Log.Logger = loggerConfiguration.CreateLogger();
+                var logger = loggerConfiguration.CreateLogger();
+
+                if (preserveStaticLogger)
+                {
+                    Log.Logger = logger;
+                }
+
+                logging.Services.AddSingleton<Serilog.ILogger>(logger);
+                logging.Services.AddSingleton(typeof(ILogger<>), typeof(Logger<>));
 
-                logging.AddConfiguration(configuration.GetSection(SystemLoggersEntryName));
-                logging.AddSerilog();
+                logging.AddConfiguration(context.Configuration.GetSection(SystemLoggersEntryName));
+                logging.AddSerilog(logger, dispose: true);
```

Comment:
I think I know why we haven't used [AddSerilog](https://github.com/serilog/serilog-extensions-hosting/blob/962e7214eb7dd5fb3b2e4347189baf5c9d260d9a/src/Serilog.Extensions.Hosting/SerilogServiceCollectionExtensions.cs#L129-L216) in the `ConfigureServices`, but let's be sure - why?